packages/mdc-tab-bar/foundation.js

```diff
@@ -0,0 +1,365 @@
+    let tabIndex = this.adapter_.getActiveTabIndex();
+
+    if (shouldGoToEnd) {
+      tabIndex = maxTabIndex;
+    } else if (shouldDecrement) {
+      tabIndex -= 1;
+    } else if (shouldIncrement) {
+      tabIndex += 1;
+    } else {
+      tabIndex = 0;
+    }
+
+    if (tabIndex < 0) {
+      tabIndex = maxTabIndex;
+    } else if (tabIndex > maxTabIndex) {
+      tabIndex = 0;
+    }
+
+    this.activateTab(tabIndex);
+  }
+
+  /**
+   * Calculates the scroll increment that will make the tab at the given index visible
+   * @param {number} index The index of the tab
+   * @param {number} nextIndex The index of the next tab
+   * @param {number} scrollPosition The current scroll position
+   * @param {number} barWidth The width of the Tab Bar
+   * @return {number}
+   * @private
+   */
+  calculateScrollIncrement_(index, nextIndex, scrollPosition, barWidth) {
+    const nextTabDimensions = this.adapter_.getTabDimensionsAtIndex(nextIndex);
+    const relativeContentLeft = nextTabDimensions.contentLeft - scrollPosition - barWidth;
+    const relativeContentRight = nextTabDimensions.contentRight - scrollPosition;
+    const leftIncrement = relativeContentRight - numbers.EXTRA_SCROLL_AMOUNT;
+    const rightIncrement = relativeContentLeft + numbers.EXTRA_SCROLL_AMOUNT;
+
+    if (nextIndex < index) {
+      return Math.min(leftIncrement, 0);
+    }
+
+    return Math.max(rightIncrement, 0);
+  }
+
+  /**
+   * Calculates the scroll increment that will make the tab at the given index visible in RTL
+   * @param {number} index The index of the tab
+   * @param {number} nextIndex The index of the next tab
+   * @param {number} scrollPosition The current scroll position
+   * @param {number} scrollContentWidth The width of the scroll content
+   * @param {number} barWidth The width of the Tab Bar
+   * @return {number}
+   * @private
+   */
+  calculateScrollIncrementRTL_(index, nextIndex, scrollPosition, scrollContentWidth, barWidth) {
+    const nextTabDimensions = this.adapter_.getTabDimensionsAtIndex(nextIndex);
+    const relativeContentLeft = scrollContentWidth - nextTabDimensions.contentLeft - scrollPosition;
+    const relativeContentRight = scrollContentWidth - nextTabDimensions.contentRight - scrollPosition - barWidth;
+    const leftIncrement = relativeContentRight + numbers.EXTRA_SCROLL_AMOUNT;
+    const rightIncrement = relativeContentLeft - numbers.EXTRA_SCROLL_AMOUNT;
+
+    if (nextIndex > index) {
+      return Math.max(leftIncrement, 0);
+    }
+
+    return Math.min(rightIncrement, 0);
+  }
+
+  /**
+   * Determines the index of the next adjacent tab
+   * @param {number} index The index of the tab
+   * @param {!MDCTabDimensions} tabDimensions Whether the next tab is left or right
+   * @param {number} scrollPosition The current scroll position
+   * @param {number} barWidth The width of the tab bar
+   * @return {number}
+   * @private
+   */
+  findTabIndexClosestToEdge_(index, tabDimensions, scrollPosition, barWidth) {
```

Comment:
I'm a little confused since the docs keep talking about the "next adjacent tab" but the method name says "closest to edge". It seems like the definition of "next" is subject to what this method finds? Is there some better way to word the docs to make it more evident what this is used for?

Reply:
Good find. I need to bring the docs in-line with the code. I was using the "next adjacent tab" beforehand but that's more ambiguous than "closest to edge". I'll update to use "closest to edge".